nndownload/downloader.py

```diff
@@ -0,0 +1,53 @@
+import re
+from concurrent.futures import ThreadPoolExecutor
+
+from Crypto.Cipher import AES
+from Crypto.Util.Padding import unpad
+from tqdm.rich import tqdm
+
+M3U8_KEY_RE = re.compile(r"((?:#EXT-X-KEY)(?:.*),?URI=\")(?P<url>.*)\",IV=0x(?P<iv>.*)")
+M3U8_MAP_RE = re.compile(r"((?:#EXT-X-MAP)(?:.*),?URI=\")(?P<url>.*)\"(.*)")
```

Comment:
We can remove those out of `nndownload` now too, just so there's no confusion when we consolidate for classes